src/docs/truffle/reference/configuration.md

```diff
@@ -51,6 +51,135 @@ Any of the following solutions will remedy this issue:
+
+<details><summary>Consider this solidity contract</summary>
+
+```solidity
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.4.25 <0.9.0;
+
+import "truffle/console.sol";
+
+contract Telephone {
+    uint public secret;
+
+    constructor(uint _secret) payable { 
+        console.log("CSTR::Telephone\n\tsecret: %o", _secret); 
+        secret = _secret;
+    }
+
+    function derivedSecret() public view returns (uint256) {
+        uint256 telephoneSecret = 100 * secret + 1;
+        console.log("The passphrase is: %o", secret);
+        console.log("Leak it a hundred times, plus one: %o", telephoneSecret);
+        return telephoneSecret; 
+    }
+}
+```
+</details>
+
+<details><summary>and this JavaScript test file</summary>
+
+```javascript
+const Telephone = artifacts.require("Telephone");
+
+const getRnd = function() {
+  const [min, max] = [13, 1013];
+  return Math.floor(Math.random() * (max - min + 1)) + min;
+}
+
+contract("Telephone", function(/* accounts */) {
+  let subj;
+  describe("Works with new deployment", function() {
+    let newSecret;
+    before(async function() {
+      newSecret = getRnd();
+      subj = await Telephone.new(newSecret);
+    });
+    
+    it("Should have a deployed contract", async function() {
+      return assert.isTrue(subj !== undefined);
+    });
+    
+    it("has the secret", async function() {
+      const best = await subj.secret();
+      return assert.equal(newSecret, best);
+    });
+    
+    it("has the derived Secret", async function() {
+      const best = await subj.derivedSecret();
+      const expected = 100 * newSecret + 1;
+      return assert.equal(expected, best);
+    });
+  });
+});
+```
+</details>
+
+#### displayPrefix
+
+You can set the `displayPrefix` string to decorate `consoleLog`'s output. This
+is useful when you want to differentiate consoleLog in the terminal.
+
+For example:
+
+```console
+module.exports = {
+  ...
+  solidityLog: {
+    displayPrefix: " :";
+  }
+}
+```
+
+Would produce the following `truffle test` output
+
+```console
+$ truffle test
+
+Compiling your contracts...
+===========================
+> Compiling ./contracts/Telephone.sol
+> Artifacts written to /tmp/test--19090-s3Aobp6MBrr3
+> Compiled successfully using:
+   - solc: 0.8.17+commit.8df45f5f.Emscripten.clang
+
+  Contract: Telephone
+    Works with new deployment
+  : CSTR::Telephone
+  :     secret: 786n
+      ✔ Should have a deployed contract
+      ✔ has the secret
+  : The passphrase is: 786n
+  : Leak it a hundred times, plus one: 78601n
+      ✔ has the derived Secret
+
+  3 passing (161ms)
+```
+
+#### preventConsoleLogMigration
+
+Because it's hard to exactly reason about gas usage when a console.log is is
+deployed to mainnet, you can opt-in to guarding your project. With
+`solidityLog.preventConsoleLogMigration` is true, `truffle migrate` will stop
+migration on mainnet.
```

Comment:
This could be worded better.

Maybe something like:

```
Because it's hard to reason about gas usage when a `console.log` is deployed, you can opt-in to guarding your project from deployments to Ethereum Mainnet that include `console.log` statements. When `solidityLog.preventConsoleLogMigration` is set to `true`, `truffle migrate` will stop
migration to Ethereum Mainnet.
```

?

Reply:
Haha saw your comment after writing my suggestion. I agree this is worded better, but also `opt-in` should be `opt in`.